src/test/java/com/hermesworld/ais/galapagos/adminjobs/impl/CreateBackUpJobTest.java

```diff
@@ -0,0 +1,188 @@
+            public Collection<TopicMetadata> getObjects() {
+                TopicMetadata meta1 = new TopicMetadata();
+                meta1.setName("topic-1");
+                meta1.setOwnerApplicationId("app-1");
+                meta1.setType(TopicType.EVENTS);
+                return List.of(meta1);
+            }
+
+        });
+        repositories.put("subscriptions", new TopicBasedRepositoryMock<SubscriptionMetadata>() {
+            @Override
+            public Class<SubscriptionMetadata> getValueClass() {
+                return SubscriptionMetadata.class;
+            }
+
+            @Override
+            public String getTopicName() {
+                return "subscriptions";
+            }
+
+            @Override
+            public Collection<SubscriptionMetadata> getObjects() {
+                SubscriptionMetadata sub1 = new SubscriptionMetadata();
+                sub1.setId("123");
+                sub1.setClientApplicationId("app-1");
+                sub1.setTopicName("topic-1");
+                sub1.setState(SubscriptionState.APPROVED);
+                return List.of(sub1);
+            }
+
+        });
+        repositories.put("application-owner-requests", new TopicBasedRepositoryMock<ApplicationOwnerRequest>() {
+            @Override
+            public Class<ApplicationOwnerRequest> getValueClass() {
+                return ApplicationOwnerRequest.class;
+            }
+
+            @Override
+            public String getTopicName() {
+                return "application-owner-requests";
+            }
+
+            @Override
+            public Collection<ApplicationOwnerRequest> getObjects() {
+                ApplicationOwnerRequest req = new ApplicationOwnerRequest();
+                req.setApplicationId("app-1");
+                req.setId("1");
+                req.setUserName("myUser");
+                req.setState(RequestState.APPROVED);
+                return List.of(req);
+            }
+
+        });
+
+        mapper = JsonUtil.newObjectMapper();
+        kafkaClusters = mock(KafkaClusters.class);
+        KafkaCluster testCluster = mock(KafkaCluster.class);
+        KafkaCluster prodCluster = mock(KafkaCluster.class);
+        when(testCluster.getId()).thenReturn("test");
+        when(prodCluster.getId()).thenReturn("prod");
+        when(testCluster.getRepositories()).thenReturn(new HashSet<>(repositories.values()));
+        when(prodCluster.getRepositories()).thenReturn(new HashSet<>(repositories.values()));
+        when(testCluster.getRepository("topics", TopicMetadata.class))
+                .thenReturn((TopicBasedRepository<TopicMetadata>) repositories.get("topics"));
+        when(testCluster.getRepository("subscriptions", SubscriptionMetadata.class))
+                .thenReturn((TopicBasedRepository<SubscriptionMetadata>) repositories.get("subscriptions"));
+        when(testCluster.getRepository("application-owner-requests", ApplicationOwnerRequest.class)).thenReturn(
```

Comment:
You could simplify that and just provide a "method" for getRepository, and use the first argument for the `repositories.get()` call. E.g.:
```
when(testCluster.getRepository(any(), any()).then(inv -> repositories.get(inv.getArgument(0)));
```

Although I am not sure that this will work because of that class parameter. If it does not work, leave it as it is now.